library/Businessprocess/IcingaDbBackend.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Icinga\Module\Businessprocess;
+
+use Icinga\Module\Businessprocess\Common\IcingadbDatabase;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Icingadb\Model\Service;
+use ipl\Orm\Compat\FilterProcessor;
+use ipl\Orm\Query;
+
+class IcingaDbBackend
+{
+    use MonitoringRestrictions;
+
+    use IcingadbDatabase;
+
+    /** @var BpConfig */
+    protected $config;
+
+    protected $backend;
+    public function __construct()
+    {
+        $this->backend = $this->getDb();
+    }
+
+    public function fetchHosts()
+    {
+
+        $hosts = Host::on($this->getDb())
```

Comment:
Use `$this->db` (or whatever variable you've cached in the constructor) instead of the call.